var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -142,6 +142,11 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     cuda_arches = spack_cuda_arch_map.values()
     conflicts("+cuda", when="cuda_arch=none")
 
+    # +cuda needs +wrapper, except when clang is compiler
+    # see spack/spack issue #20347
+    conflicts("+cuda", when="%gcc ~wrapper")
+    conflicts("+cuda", when="%intel ~wrapper")
```

Comment:
Apologies but I was not sufficiently clear: for this particular approach to work, each non-clang compiler listed in `lib/spack/spack/compilers/` would need an entry here. What you have probably covers the common cases but if a user tries to install with `%pgi` or `%xl` these `conflicts` statements would not catch the issue.

While this is better than what exists now, we have generally tried to be as precise as possible when detecting problems. I suggested adding a modified install method because you could just check for `not satisfies(%clang)`. Would you be able to try that?

Reply:
Defining an `install` method as you suggest doesn't stop the install from proceeding. In fact it looks like the code in it isn't even executed. ??

Reply:
To be clear, my thought was that it would end up trying to install all dependencies, then fail when installing `kokkos`, are you saying the `kokkos` install itself proceeds?

Reply:
yes, kokkos itself proceeds to the original error